OpenDreamRuntime/DreamThread.cs

```diff
@@ -86,16 +86,32 @@ public abstract class DreamProc {
         }
     }
 
-    sealed class CancellingRuntime : Exception {
-        public CancellingRuntime(string message)
-            : base(message)
-        {}
+    [Virtual]
+    class DMRuntime : Exception {
+        public readonly DreamValue Value;
+        public DMRuntime(string message)
+            : base(message) {
+            Value = new DreamValue(message);
+        }
+
+        public DMRuntime(DreamValue value) : base($"'throw' thrown ({value})") {
+            Value = value;
+        }
+    }
+
+    sealed class DMCrashRuntime : DMRuntime {
+
+        public DMCrashRuntime(string message) : base(new DreamValue(message)) { }
```

Comment:
Using this base constructor will end up putting `'throw' thrown` at the beginning of the exception message. Probably meant to use the other one?
```suggestion
        public DMCrashRuntime(string message) : base(message) { }
```